MapboxCoreNavigation/Bundle.swift

```diff
@@ -2,15 +2,21 @@ import Foundation
 
 extension Bundle {
     
-    var backgroundModeLocationSupported: Bool {
-        get {
-            if let modes = object(forInfoDictionaryKey: "UIBackgroundModes") as? [String] {
-                return modes.contains("location")
+    struct BackgroundModes {
+        var location: Bool {
+            return backgroundModesContains("location")
+        }
+        
+        fileprivate func backgroundModesContains(_ key: String) -> Bool {
```

Comment:
For a more natural representation, `BackgroundModes` could be a string-backed `OptionSet` or a `Set<String>`, and `backgroundModes` could be a lazily variable that reads from the info dictionary.